### PR TITLE
Add validation.yml schema, expand examples

### PR DESCRIPTION
**Pull request by apexskier**
The new schema helps local dev and documentation, since an editor can provide autocompletions and inline help if configured properly. Depending on future implementation, it can also help catch mistakes in contribution format that could mess up how the site is built.

The bulk of the code changes are expanding "examples" support. Right now collection types (array and dict) don't really work - the "title" is "Set XXX", but the current use cases do `array-add` which appends to the existing list. I've added shell escaping to values, which allows using quotes or multiline strings (writing an entire dictionary with quotes in the contents, for example - https://github.com/yannbertrand/macos-defaults/pull/146).

I've also made some minor tweaks to the generated pages to be more semantically correct (don't use a block quote for the first line of the description or a list for the first line of an example's text, etc).

I'd like some feedback on this, as I'm new to the project.

A few notes from a newcomer's perspective:

- Having five build systems makes contributing non-trivial changes hard, as there's a lot more code to search through and update. I've only made changes to support the production one.
- Having an example for "default" doesn't make a lot of sense to me. I'd prefer to just document _what_ the default is and inform people that running `defaults delete <domain> <key>` will reset to that default.
- I have mixed feelings about dynamically generating example shell commands - I like that it enforces a format, but it's pretty complex and might limit some use case.

### CODE REVIEW

File: defaults-fr.yml [hunk 1]
```diff
@@ -236,7 +236,7 @@ categories:
             default: true
             text: Screenshot 2020-01-09 at 13.27.20.png
           - value: false
-            text:
+            text: |
```
Comment (yannbertrand) [inline defaults-fr.yml:239]: I'm not sure on why this is necessary?

File: defaults-fr.yml [hunk 2]
```diff
@@ -545,11 +545,18 @@ categories:
...
           Ajouter des suffixes supplémentaires que Xcode place dans le menu "Related Items > Counterparts".
         param:
-          type: array-add
+          type: array
         examples:
-          - value: '"ViewModel" "View"'
+          - command: array-add
+            value:
+              - ViewModel
+              - View
```
Comment (yannbertrand) [inline defaults-fr.yml:553]: This is great 👍

File: build/production/__snapshots__/write-pages.test.js.snap [hunk 3]
```diff
@@ -19,40 +19,42 @@ meta:
...
-> Page description.
+Page description.
 
 - **Tested on macOS**:
   * Big Sur
 - **Parameter type**: bool
 
 ## Set to \`true\` (default value)
+
+
```
Comment (yannbertrand) [inline build/production/__snapshots__/write-pages.test.js.snap:30]: I'm beeing fussy but I would prefer to only get one empty line

**Review by yannbertrand — CHANGES_REQUESTED** [review]
Welcome and thanks a lot, this is awesome!

I did some graphical choices that in fact are not semantically correct.
Everything you've pointed out are great points.

About your last notes:
- Do not bother with the other build folders, that was a technical trial which is not the point of this repo. I'm still not sure on keeping it under this one.
- I'm not sure to understand this one. I agree that the "Delete current value" title is less clear than "Reset to default value"
- 100% agreed, do you think we could improve by switching to a full command in the `examples` section instead of the command name?

File: defaults-fr.yml (hunk 1 above)
Comment (apexskier) [reply defaults-fr.yml:239]: The current yaml syntax is a list of strings, while the schema expects a string. This turns it into a normal string (which is then processed as markdown). I'm interpreting the intent here, might have gotten something wrong.

**Comment by apexskier** [comment]
> I'm not sure to understand this one. I agree that the "Delete current value" title is less clear than "Reset to default value"

I'm not entirely sure what I'd prefer either. I don't fully understand where no value is a default or if a tool sets a value by default within the defaults system. Just a thought, I don't have any other changes I want to make right now.

> 100% agreed, do you think we could improve by switching to a full command in the examples section instead of the command name?

Yeah, that's what I'm thinking. At this point I think I'm happy enough with the generation, given what I've done in this PR. 

--- 

One note - I might be a bit delayed following up here. My computer died on me just as I was finishing up this stuff and won't let me past the login screen so I'm doing an OS reinstall.

File: build/production/__snapshots__/write-pages.test.js.snap (hunk 3 above)
Comment (yannbertrand) [reply build/production/__snapshots__/write-pages.test.js.snap:30]: I'll fix it in another branch

**Comment by yannbertrand** [comment]
> One note - I might be a bit delayed following up here. My computer died on me just as I was finishing up this stuff and won't let me past the login screen so I'm doing an OS reinstall.

Good luck with that!

**Review by yannbertrand — APPROVED** [review]

**Comment by yannbertrand** [comment]
Thanks a lot, I'm rewriting history and will merge it in master :)

**Comment by yannbertrand** [comment]
It's online!

**Pull request merged**